pmd-core/src/main/java/net/sourceforge/pmd/util/document/DocumentFile.java

```diff
@@ -36,26 +34,42 @@
     private final BufferedReader reader;
     private int currentPosition = 0;
 
-    private final Path temporaryPath = Files.createTempFile("pmd-", null);
-    private final Writer writer = new FileWriter(temporaryPath.toFile());
+    private final Path temporaryPath = Files.createTempFile("pmd-", ".tmp");
+    private final Writer writer = Files.newBufferedWriter(temporaryPath, StandardCharsets.UTF_8);
```

Comment:
the charset assumption may be wrong

Reply:
Is there anywhere I can get the charset for the file i'm editing?

Reply:
PMD itself takes it from the user input (CLI, Gradle, Maven, etc.), we should probably honor that. Just have both encodings be the same, and taken at the constructor, so you can hook it to the current execution configuration at the proper layer.